torch/_prims/__init__.py

```diff
@@ -1832,19 +1833,38 @@ def _as_strided_scatter_meta(
 #
 # Shape operations
 #
-def collapse(a: Tensor, start: int, end: int) -> Tensor:
-    """
-    Wrapper around reshape that collapses a span of dimensions.
 
-    See collapse_view for the corresponding view operation.
-    """
 
-    dim_length = 1
-    for idx in range(start, end):
-        dim_length = dim_length * a.shape[idx]
+def _collapse_meta(a: Tensor, start: int, end: int) -> Tensor:
+    # Special-case for zero dimensional tensors
+    ndim = max(1, a.dim())
+    utils.validate_idx(ndim, start)
+    utils.validate_exclusive_idx(ndim, end)
```

Comment:
Won't this prevent `collapse(t, 0, 0)` - is that invalid?

Reply:
For whatever reason `aten.flatten(t, 0, 1)` flattens dims [0, 1] inclusive while `prims.collapse(t, 0, 1)` collapses dims `[0, 1)` exclusive. So the range [0, 0) is empty which is not allowed.

Reply:
Why does behavior of `prims.collapse` and `aten.flatten` differ? We are under no obligation to follow `lax` names and behavior, or are we?

Reply:
Looks like `prims.collapse` has been this way since the first prims PR #75095. cc @mruberry

Reply:
We didn't intend to fix the prims' from day one, I was just modeling them on lax at the time to get the project started